File: Daily/sbgo/Leetcode92-反转链表.go
package sbgo

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseBetween(head *ListNode, left int, right int) *ListNode {
	dummy := &ListNode{Val: -1}
	dummy.Next = head
	a := dummy
	for i := 0; i < left-1; i++ {
		a = a.Next
	}
	b := a.Next
	c := b.Next
	for i := 0; i < right-left; i++ {
		d := c.Next
		c.Next = b
		b, c = c, d
	}
	a.Next.Next = c
	a.Next = b
	return dummy.Next
}

File: Leetcode/PY/Leetcode92-反转链表.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def reverseBetween(self, head: ListNode, left: int, right: int) -> ListNode:
        dummy = ListNode(-1)
        dummy.next = head
        a = dummy
        for i in range(left - 1):
            a = a.next
        b  = a.next
        c = b.next
        for i in range(right - left):
            d = c.next
            c.next = b
            b,c = c,d
        a.next.next = c
        a.next  = b
        return dummy.next
        